atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/WriteOnceDeleteOnceWorkflows.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.workflow;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.palantir.atlasdb.buggify.api.BuggifyFactory;
+import com.palantir.atlasdb.buggify.impl.DefaultBuggifyFactory;
+import com.palantir.atlasdb.buggify.impl.DefaultNativeSamplingSecureRandomFactory;
+import com.palantir.atlasdb.workload.store.ImmutableWorkloadCell;
+import com.palantir.atlasdb.workload.store.InteractiveTransactionStore;
+import com.palantir.atlasdb.workload.store.WorkloadCell;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import java.security.SecureRandom;
+import java.util.Optional;
+import java.util.function.IntFunction;
+
+/**
+ * The idea of this class is to catch bugs around tombstone drops, by having a
+ * workflow which _mostly_ writes once then deletes once for a given cell. Probability and randomness are used to
+ * enforce this behaviour, thus it's possible to have a write without a delete, or visa versa; some cells may
+ * randomly be written or deleted multiple times as well.
```

Comment:
do we want to be more specific (initial transactions _mostly_ write, later transactions _mostly_ delete)?